utilities/backupable/backupable_db.cc

```diff
@@ -730,8 +731,22 @@ Status BackupEngineImpl::Initialize() {
 
   // set up threads perform copies from files_to_copy_or_create_ in the
   // background
+  threads_.reserve(options_.max_background_operations);
+  thread_ids_.resize(options_.max_background_operations);
+  port::Mutex mu;
+  port::CondVar cv(&mu);
+  size_t size = 0;
   for (int t = 0; t < options_.max_background_operations; t++) {
-    threads_.emplace_back([this]() {
+    threads_.emplace_back([&mu, &cv, &size, t, this]() {
+      auto id = port::GetCurrentThreadId();
+      {
+        MutexLock l(&mu);
+        thread_ids_[t] = id;
+        size++;
+        if (size == thread_ids_.size()) {
+          cv.SignalAll();
+        }
+      }
```

Comment:
This feels complicated. Is it possible to pass `options.enable_update_background_thread_cpu_priority` all the way to here and it just call `port::SetCpuPriority(0, 19);`?

Reply:
these threads are set up during initialization of `BackupEngine`, `options` is passed in for each `CreateNewBackup` after construction of `BackupEngine`,  so I don't think it's possible to pass in the options here.

But once we switch to using [pthread_setschedparam](https://linux.die.net/man/3/pthread_setschedparam) to set both sched type and priority, we don't need the thread ids, and can get rid of this logic here.